src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/halapi.cxx

```diff
@@ -659,21 +664,23 @@ void*
 xclGraphOpen(xclDeviceHandle handle, const uuid_t xclbin_uuid, const char* graph, xrt::graph::access_mode am)
 {
   try {
-    xclcpuemhal2::CpuemShim *drv = xclcpuemhal2::CpuemShim::handleCheck(handle);
-    xclGraphHandle graphHandle = new xclcpuemhal2::GraphType(drv, graph);
+    std::shared_ptr<xclcpuemhal2::CpuemShim> drv{xclcpuemhal2::CpuemShim::handleCheck(handle)};
```

Comment:
handleCheck returns a ptr* (good), but the `handle` is deleted `delete _M_ptr` when the drv shared_ptr goes out scope, is that really what you want?  The `drv` is passed to the constructor of GraphType, but converted to a weak_ptr.    Something is not right here/

Reply:
The shim handles are added to global maps when they are created. (xclOpen/ xclGraphOpen)
 std::map<unsigned int, std::shared_ptr<CpuemShim> > devices;
 std::map<unsigned int, std::shared_ptr<GraphType> > graph_devices;
So the local shared pointers are used to access the instance and then to be destructed. The global map deletes the instances once xclClose/xclGraphClose calls are received.

Reply:
@sgundime-xilinx Please check again.  handleCheck returns a raw ptr, you construct a scoped shared ptr from this raw pointer.   The lifetime of the raw pointer is tied to shared ptr constructed on line 667.   The std::map you are referring to will contain a shared ptr wrapping dead memory if the shared ptr on line 667 goes out of scope.  If you had a shared from this, or if handleCheck returned a shared ptr, then there would be no problem.    Are you really sure this function is correct?

Reply:
@stsoe you are right! I wonder how it missed in canary test cases. This comment helped me in adding PS enabled AIE sw_emu test case. I used raw handler pointer only now which resolved the problem.

Reply:
The problem with XRT is that it is not well tested outside of the copy-paste tutorial-like examples.
This is why we hit so many problems with our SYCL work-flow because the XRT usage patterns are quite unusual compared to tested use cases.
I hope this PR will not add more bugs to our SYCL implementation than we had before.

Reply:
I think this PR is high risk despite the PR description claiming otherwise; the changes are overly complicated IMO.